src/utils/asset.js

```diff
@@ -135,6 +142,131 @@ export const getExplorerTransactionHash = (asset, hash) => {
   switch (asset) {
     case 'NEAR':
       return hash.split('_')[0]
-    default: return hash
+    default:
+      return hash
   }
 }
+
+export const TOKEN_DETAILS = {
+  ethereum: {
+    sourceUrl(contractAddress) {
+      return `https://api.ethplorer.io/getTokenInfo/${contractAddress}?apiKey=freekey`
+    },
+    async getDetails(contractAddress) {
+      const data = await fetchData(this.sourceUrl(contractAddress))
+
+      const { symbol, name, decimals } = data
+
+      return { symbol, name, decimals }
+    }
+  },
+  bsc: {
+    sourceUrl(contractAddress) {
+      return `https://wallet.binance.org/api/v1/bsc-mainnet/assets/${contractAddress}`
+    },
+    async getDetails(contractAddress) {
+      const data = await fetchData(this.sourceUrl(contractAddress))
+
+      const { displaySymbol, name, decimals } = data
+
+      return { symbol: displaySymbol, name, decimals }
+    }
+  },
+  polygon: {
+    sourceUrl(contractAddress) {
+      return {
+        symbol: `https://api.polygonscan.com/api?module=proxy&action=eth_call&to=${contractAddress}&data=0x06fdde03&tag=latest&apikey=8XR2CZ5HY3JRP6J4BGP1TBC3WJEU88NSVZ`,
+        name: `https://api.polygonscan.com/api?module=proxy&action=eth_call&to=${contractAddress}&data=0x95d89b41&tag=latest&apikey=8XR2CZ5HY3JRP6J4BGP1TBC3WJEU88NSVZ`,
+        decimals: `https://api.polygonscan.com/api?module=proxy&action=eth_call&to=${contractAddress}&data=0x313ce567&tag=latest&apikey=8XR2CZ5HY3JRP6J4BGP1TBC3WJEU88NSVZ`
+      }
+    },
+    async getDetails(contractAddress) {
+      const [_symbol, _name, _decimals] = await Promise.all([
+        fetchData(this.sourceUrl(contractAddress).symbol),
+        fetchData(this.sourceUrl(contractAddress).name),
+        fetchData(this.sourceUrl(contractAddress).decimals)
+      ])
+      return parseAssetData(_symbol.result, _name.result, _decimals.result)
+    }
+  },
+  rsk: {
+    sourceUrl() {
+      return 'https://public-node.rsk.co'
+    },
+    rpcBody(contractAddress) {
```

Comment:
I'm wondering, if we are doing like this for some of them, Why don't we just use this method for everything so there is one single implementation? We can keep it very tidy by using ethers too.

What is the benefit of using the apis?